File: python/06_sum_square_difference.py
def sum_square_difference(n: int) -> int:
    
    sum_squares = 0
    sum = n * (n + 1) // 2
    
    for i in range(1, n + 1):
        sum_squares += i * i
    
    return abs(sum * sum - sum_squares)
    
if __name__ == "__main__":
    print("Difference between sum of the squares of first 100 natural numbers and square of the sum:", sum_square_difference(100))

File: cpp/06_sum_square_difference.cpp
#include <iostream>

int sum_square_difference(int n) {
    
    // summation formula
    int sum = n * (n + 1) / 2;
    
    // calculate sum of squares
    int sum_squares = 0;
    for (int i = 1; i <= n; i++){
        sum_squares += i*i;
    }
    
    return sum*sum - sum_squares;
}

int main() {
    std::cout << sum_square_difference(100) << "\n";
}
